src/CodeGen_LLVM.cpp

```diff
@@ -550,47 +562,43 @@ void CodeGen_LLVM::compile_func(const LoweredFunc &f) {
     {
         size_t i = 0;
         for (auto &arg : function->args()) {
-            sym_push(args[i].name, &arg);
-            if (args[i].is_buffer()) {
-                push_buffer(args[i].name, &arg);
+            sym_push(f.args[i].name, &arg);
+            if (f.args[i].is_buffer()) {
+                push_buffer(f.args[i].name, &arg);
             }
 
             i++;
         }
     }
+}
 
-    // Ok, we have a module, function, context, and a builder
-    // pointing at a brand new basic block. We're good to go.
-    f.body.accept(this);
-
+void CodeGen_LLVM::end_func(const LoweredFunc &f) {
     return_with_error_code(ConstantInt::get(i32, 0));
 
     // Remove the arguments from the symbol table
-    for (size_t i = 0; i < args.size(); i++) {
-        sym_pop(args[i].name);
-        if (args[i].is_buffer()) {
-            pop_buffer(args[i].name);
+    for (size_t i = 0; i < f.args.size(); i++) {
+        sym_pop(f.args[i].name);
+        if (f.args[i].is_buffer()) {
+            pop_buffer(f.args[i].name);
         }
     }
 
-    module->setModuleIdentifier("halide_module_" + name);
+    module->setModuleIdentifier("halide_module_" + f.name);
```

Comment:
We're already setting the module identifier in compile(Module) , I think this is just stale code that should be removed.